light/rpc/client.go

```diff
@@ -579,8 +579,16 @@ func (c *Client) Tx(ctx context.Context, hash []byte, prove bool) (*ctypes.Resul
 		return nil, err
 	}
 
-	// Validate the proof.
-	return res, res.Proof.Validate(l.DataHash)
+	// Check if the proof is correctly constructed.
+	if err := res.Proof.Validate(); err != nil {
+		return nil, err
+	}
+
+	// Verify if the proof is valid
```

Comment:
[nit] I would refrain from using `valid` to describe proofs that have been verified. Instead you could use:

- If a proof passes `Validate` then it is valid
- If a proof passes `Verify` then it is verified

```suggestion
	// Verify the proof
```